src/gage.c

```diff
@@ -556,6 +567,8 @@ int getNextRainfall(int j)
             }
             else return 0;
         }
+
```

Comment:
@abhiramm7 Looks like you need to add the condition here for the next rainfall supplied from the API. 

```C
else if (Gage[j].dataSource == RAIN_API)
{
    rNext = Gage[j].externalRain;
}
```

The function `getNextRainfall(int j)` is what provides each raingauge with the next rainfall rate in `ft/sec`.

Reply:
When I added that simulation is not executing. I am not sure what is happening there. Simulation is getting stuck in the first time step. Hence, I had to push it to the other function.

Reply:
@abhiramm7, I see how that can happen now.  

Why don't you add a conditional before the do - while loop.


```c
Gage[j].nextRainfall = 0.0;
if rain_api:
    set value
else
    the do - while loop as it was before
```

Reply:
#solvedinfiniteloop :-)

Reply:
Ah, I'll give it a shot

Reply:
Done, updated. It works !